Source/GSStoryboardTransform.m

```diff
@@ -0,0 +1,878 @@
+
+#import <Foundation/NSData.h>
+#import <Foundation/NSDictionary.h>
+#import <Foundation/NSString.h>
+#import <Foundation/NSMapTable.h>
+#import <Foundation/NSXMLDocument.h>
+#import <Foundation/NSXMLNode.h>
+#import <Foundation/NSXMLElement.h>
+#import <Foundation/NSUUID.h>
+#import <Foundation/NSBundle.h>
+#import <Foundation/NSKeyedArchiver.h>
+#import <Foundation/NSException.h>
+
+#import "AppKit/NSSeguePerforming.h"
+#import "AppKit/NSStoryboard.h"
+#import "AppKit/NSStoryboardSegue.h"
+#import "AppKit/NSNibDeclarations.h"
+#import "AppKit/NSViewController.h"
+#import "AppKit/NSWindowController.h"
+
+#import "GSStoryboardTransform.h"
+#import "GSFastEnumeration.h"
+
+#define APPLICATION @"application"
+
+@interface NSStoryboardSegue (__private__)
+- (void) _setDestinationController: (id)controller; 
+- (void) _setSourceController: (id)controller;
+@end
+
+@interface NSStoryboardSegue (__StoryboardPrivate__)
+// Private to this class...
+- (void) _setKind: (NSString *)k;
+- (void) _setRelationship: (NSString *)r;
+- (NSString *) _kind;
+- (NSString *) _relationship;
+@end
+
+// this needs to be set on segues
+@implementation NSStoryboardSegue (__StoryboardPrivate__)
+- (void) _setKind: (NSString *)k
+{
+  ASSIGN(_kind, k);
+}
+
+- (void) _setRelationship: (NSString *)r
+{
+  ASSIGN(_relationship, r);
+}
+
+- (NSString *) _kind
+{
+  return _kind;
+}
+
+- (NSString *) _relationship
+{
+  return _relationship;
+}
+@end
+
+@implementation NSStoryboardSeguePerformAction
+- (id) target
+{
+  return _target;
+}
+
+- (void) setTarget: (id)target
+{
+  ASSIGN(_target, target);
+}
+
+- (SEL) action
+{
+  return _action;
+}
+
+- (void) setAction: (SEL)action
+{
+  _action = action;
+}
+
+- (NSString *) selector
+{
+  return NSStringFromSelector(_action);
+}
+
+- (void) setSelector: (NSString *)s
+{
+  _action = NSSelectorFromString(s);
+}
+
+- (id) sender
+{
+  return _sender;
+}
+
+- (void) setSender: (id)sender
+{
+  ASSIGN(_sender, sender);
+}
+
+- (NSString *) identifier
+{
+  return _identifier;
+}
+
+- (void) setIdentifier: (NSString *)identifier
+{
+  ASSIGN(_identifier, identifier);
+}
+
+- (NSString *) kind
+{
+  return _kind;
+}
+
+- (void) setKind: (NSString *)kind
+{
+  ASSIGN(_kind, kind);
+}
+
+- (NSStoryboard *) storyboard
+{
+  return _storyboard;
+}
+
+- (void) setStoryboard: (NSStoryboard *)storyboard
+{
+  ASSIGN(_storyboard, storyboard);
+}
+
+- (NSStoryboardSegue *) storyboardSegue
+{
+  return _storyboardSegue;
+}
+
+- (void) setStoryboardSegue: (NSStoryboardSegue *)ss
+{
+  ASSIGN(_storyboardSegue, ss);
+}
+
+- (void) dealloc
+{
+  RELEASE(_storyboard);
+  RELEASE(_kind);
+  RELEASE(_identifier);
+  RELEASE(_sender);
+  RELEASE(_storyboardSegue);
+  [super dealloc];
+}
+
+- (IBAction) doAction: (id)sender
+{
+  BOOL should = YES;
+  BOOL responds = [_sender respondsToSelector: @selector(shouldPerformSegueWithIdentifier:sender:)];
+
+  if (responds)
+    {
+      should = [_sender shouldPerformSegueWithIdentifier: _identifier
```

Comment:
Why do you treat this method as optional whereas the other two from the protocol are handled as being required?

Reply:
In the NSStoryboard_test example, there Is a special case where a segue can be invoked from a non -controller (i.e. an NSMenuItem).  So, on those objects, that method won't be there.

Reply:
I just now pushed a check for this.